### ISSUE TITLE
JAWS screen reader doesn't read the content of popup windows

### ISSUE THREAD

**Issue by olgachizhevskiy** [wet-boew/wet-boew#6427]
Hello,

I'm trying to read the following page with JAWS 16.0 screen reader: http://wet-boew.github.io/v4.0-ci/demos/lightbox/lightbox-en.html

I'm trying to open the links under "Single Items" header. 
When I click on the first image, then the popup with the image is opened and the reader reads "Image 1" and some other stuff.
But when I click on "AJAX - Example 1" link, then popup window is shown, but reader says only one word "document" and doesn't read the content of the popup.

How can I make the content of the popup to be read by screen reader?

Regards,
Olga

**Comment by LaurentGoderre** [comment]
@dfarough can you give any insight on this?

**Comment by pjackson28** [comment]
That's a known issue for Jaws where it doesn't read the first nested header within a region like NVDA does (and is supposed to). The content is still there and can be read by instructing Jaws to read but it won't be read off automatically without user intervention. In PR #6430 I added an aria-labelledby to the Lightbox content area to make that explicit association so it gets read automatically.

**Comment by olgachizhevskiy** [comment]
Hello,

 

Could you please help me to fix it in my code for older version of plugin?

 

This is html of my popup that opens by clicking on the link:

 

<div id="divClueText" class="mfp-hide white-popup-block">

            <div class="clueContent">

            <center>

            <div class="clueHeader">_Clue_:</div>

            [clueText]

            </center>

            </div>

</div>

 

Regards,

Olga

**Comment by LaurentGoderre** [comment]
It's not recommended to hard code this and it would be preferable to update to either the latest or wait until v4.0.10 on January 20th

**Comment by olgachizhevskiy** [comment]
Hello, 

 

I’ve just tried to open http://wet-boew.github.io/v4.0-ci/demos/lightbox/lightbox-en.html and read it with JAWS. I suppose it has the latest version.

 

Now JAWS reads only the title of the popup, but not all the content.

 

Regards,

Olga

**Comment by LaurentGoderre** [comment]
I would strongly caution about what the desired behavior should be if you are not an everyday user of JAWS. Whether this is an issue should be up to actual users.

**Comment by olgachizhevskiy** [comment]
I’m not an everyday user of JAWS, I’m a developer. I have popup window on my page and content of the window is not read  when the popup window is opened. I got this bug from the client and I need to fix it now. And your site I’ve used just as example of the problem.

 

Could you please help me with this?

**Comment by pjackson28** [comment]
@olgachizhevskiy NVDA reads the content automatically yet for whatever reason Jaws doesn't (stops reading automatically once it hits a document node). There really isn't anything we could do about that short of applying aria-describedby but that would force NVDA to hear the content twice and can cause Jaws to crash if there is a lot of content (so the fix would be worse than the issue). The ultimate impact for Jaws users with it not reading off automatically would be negligible since the title is read off and the content is still available through any of the normal commands for reading the content (and Jaws does announce that it is at a document node).

**Comment by olgachizhevskiy** [comment]
So, the best solution for JAWS would be even not to add this aria-describedby attribute since it makes problems in other screen readers, but just to stay with initial “document” word read? 

.  <https://github.com/notifications/beacon/AGKayt0ysD2fS5BDzp3GfQu-aMpGDK-gks5ngDU3gaJpZM4DPtCZ.gif>

**Comment by LaurentGoderre** [comment]
From what I understand the fix in place make sure that the title is read. Chances are that JAWS user are used to that quirk and will know what is happening and will use the command to read on.

**Comment by pjackson28** [comment]
Yes, I agree with @LaurentGoderre. The aria-labelledby fix that was introduced in the PR should be enough help. Anything we do beyond that will do more harm than good for both NVDA and Jaws users.

**Comment by olgachizhevskiy** [comment]
Thank you all for your answers.

**Comment by dfarough** [comment]
I am an experienced Jaws user.  I have tried the Ajax popup example and Jaws spoke the document header and I was able to use the normal Jaws functionality to read the content within the popup window.  Although the content is not read automatically, I do not feel that this poses a major problem since I would expect that anyone would probably use the cursor keys to determine what other content mighht be present within the dialogue.  The example seems to work fine for me.

**Comment by LaurentGoderre** [comment]
Thanks a lot @dfarough!
